### PR TITLE
Overload gdb.execute return type

### PR DESCRIPTION
**Pull request by peace-maker**
The `to_string` argument dictates the returned type. The [docs say](https://sourceware.org/gdb/current/onlinedocs/gdb.html/Basic-Python.html#Basic-Python) that if it's set to `True` it will always return a `str`. The current return type of `str | None` requires checks for None before being able to use the return value even tough it can never be None in practice.

[The code](https://sourceware.org/git/?p=binutils-gdb.git;a=blob;f=gdb/python/python.c;h=91636ef022529558dbd3e69527cdcfdc9a9c6369;hb=e53a8e8685685c97588f8319d993ea6cd5635e47#l692) does indeed always return a string when `to_string` is true.

I hope I've applied the overload pattern correctly here!

### CODE REVIEW

File: stubs/gdb/gdb/__init__.pyi [hunk 1]
```diff
@@ -23,7 +23,12 @@ STDOUT: int
 STDERR: int
 STDLOG: int
 
-def execute(command: str, from_tty: bool = ..., to_string: bool = ...) -> str | None: ...
+@overload
+def execute(command: str, from_tty: bool = False, to_string: Literal[True] = True) -> str: ...
+@overload
+def execute(command: str, *, to_string: Literal[False]) -> None: ...
+@overload
+def execute(command: str, from_tty: bool = False, to_string: bool = False) -> str | None: ...
```
Comment (srittau) [inline stubs/gdb/gdb/__init__.pyi:31]: The default is `False`, so the overloads should be:

```suggestion
@overload
def execute(command: str, *, to_string: Literal[True]) -> str: ...
@overload
def execute(command: str, from_tty: bool, to_string: Literal[True]) -> str: ...
@overload
def execute(command: str, from_tty: bool = False, to_string: bool = False) -> str | None: ...
```

That said, doesn't it always return `None` if `to_string` is `False` (or unset)? In this case, the last overload could use `Literal[False]` and return just `None`.

Comment (peace-maker) [reply stubs/gdb/gdb/__init__.pyi:31]: Thank you! I thought the last overload is required when passing a variable to `to_string` so you don't know statically if it's True or False? Maybe I'm misunderstanding.

Comment (srittau) [reply stubs/gdb/gdb/__init__.pyi:31]: I can never remember this, but I think type checkers should be able to combine overloads in that case. But it's also possible that we need a fourth overload, i.e. one with `Literal[False]` and one with `bool`.

Comment (peace-maker) [reply stubs/gdb/gdb/__init__.pyi]: I've tested with 
```python
from __future__ import annotations
from typing import overload
from typing_extensions import Literal


@overload
def execute(command: str, from_tty: bool = False, to_string: Literal[False] = False) -> None: ...
@overload
def execute(command: str, *, to_string: Literal[True]) -> str: ...
@overload
def execute(command: str, from_tty: bool, to_string: Literal[True]) -> str: ...
@overload
def execute(command: str, from_tty: bool = False, to_string: bool = False) -> str | None: ...

reveal_type(execute("bt"))
reveal_type(execute("bt", to_string=False))
reveal_type(execute("bt", from_tty=True, to_string=False))
reveal_type(execute("bt", from_tty=False, to_string=False))
reveal_type(execute("bt", from_tty=True))
reveal_type(execute("bt", from_tty=False))
reveal_type(execute("bt", to_string=True))
reveal_type(execute("bt", from_tty=True, to_string=True))
reveal_type(execute("bt", from_tty=False, to_string=True))

def some_function(thing: bool) -> None:
    reveal_type(execute("bt", to_string=thing))
    reveal_type(execute("bt", from_tty=True, to_string=thing))
    reveal_type(execute("bt", from_tty=False, to_string=thing))
```

and mypy prints the expected types
```
$ mypy /tmp/execute.pyi
/tmp/execute.pyi:15: note: Revealed type is "None"
/tmp/execute.pyi:16: note: Revealed type is "None"
/tmp/execute.pyi:17: note: Revealed type is "None"
/tmp/execute.pyi:18: note: Revealed type is "None"
/tmp/execute.pyi:19: note: Revealed type is "None"
/tmp/execute.pyi:20: note: Revealed type is "None"
/tmp/execute.pyi:21: note: Revealed type is "builtins.str"
/tmp/execute.pyi:22: note: Revealed type is "builtins.str"
/tmp/execute.pyi:23: note: Revealed type is "builtins.str"
/tmp/execute.pyi:26: note: Revealed type is "Union[builtins.str, None]"
/tmp/execute.pyi:27: note: Revealed type is "Union[builtins.str, None]"
/tmp/execute.pyi:28: note: Revealed type is "Union[builtins.str, None]"
Success: no issues found in 1 source file
```

So both overloads, one with Literal[False] and one with bool, are required it seems.

**Review by srittau — APPROVED** [review]
Thanks!

**Pull request merged**